src/UseCases/ApplyForMembership/ApplyForMembershipUseCase.php

```diff
@@ -60,7 +45,17 @@ public function applyForMembership( ApplyForMembershipRequest $request ): ApplyF
 			return ApplyForMembershipResponse::newFailureResponse( $validationResult );
 		}
 
-		$application = $this->newApplicationFromRequest( $request );
+		$paymentCreationResponse = $this->createPaymentUseCase->createPayment( $this->extractPaymentRequestDataForMembershipApplication( $request ) );
+		if ( $paymentCreationResponse instanceof FailureResponse ) {
+			// TODO what to return?
```

Comment:
Return a failure response. You can create an instance of that class with `[ApplicationValidationResult::SOURCE_PAYMENT => $paymentCreationResponse->getMessage]`. We're losing some granularity there (i.e. which fields in the form had an error), but I think that granularity is not needed any more, because the only thing that's needed at thist point is fee validation. The fact that an interval (and possibly a payment type) must be selected should be validated by the frontend, so if the response is "invalid interval" or "invalid payment type", then we can just pass that error message back to the frontend.